tests/components/mqtt/test_device_trigger.py

```diff
@@ -351,6 +348,102 @@ async def test_if_fires_on_mqtt_message(
     assert calls[1].data["some"] == "long_press"
 
 
+async def test_non_unique_triggers(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    calls: list[ServiceCall],
+    mqtt_mock_entry: MqttMockHAClientGenerator,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test non unique triggers.
+
+    The last received config for a non unique trigger is updating.
```

Comment:
This is somewhat odd, should we maybe warn or fail instead?

Reply:
Triggers can be initiated in two ways:
- From an automation
- At discovery
If a trigger does not exist, we create it. With old behavior the discovery_id was leading to make a trigger unique.
I suggest we fail discovery if there is an existing trigger that has `discovery_id` set (it is set to `None` when set up from an automation)
When updating, we fail if there is an existing trigger that has `discovery_id` set to a different id.

Reply:
Added validation, updated tests